api/datasets.go

```diff
@@ -220,31 +276,69 @@ func (h *DatasetHandlers) listHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *DatasetHandlers) getHandler(w http.ResponseWriter, r *http.Request) {
-	args, err := parseGetReqArgs(r, strings.TrimPrefix(r.URL.Path, "/get/"))
-	if err != nil {
-		util.RespondWithError(w, err)
+	params := lib.GetParams{}
+	args := &GetReqArgs{}
+	err := parseProxyParams(r, &params)
+	if err != nil && err.Error() == "bad request" {
+		util.WriteErrResponse(w, http.StatusBadRequest, err)
 		return
 	}
+	if err != nil {
+		args, err = parseGetReqArgs(r, strings.TrimPrefix(r.URL.Path, "/get/"))
```

Comment:
The gist of the approach is to try to parse the full params as they would be passed down into lib, if that doesn't work revert to the default flow handling.